src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -164,7 +167,14 @@ MlirConvArgs MakeMlirConvArgs(const std::vector<size_t>& in_dims,
     StridedMemRef5D output{nullptr, nullptr, 0, {0, 0, 0, 0, 0}, {0, 0, 0, 0, 0}};
     initDimStrides(out_dims, out_strides, output);
 
-    return {filter, input, output};
+    if(populateWorkspaceArg)
+    {
+        StridedMemRef5D workspace{nullptr, nullptr, 0, {0, 0, 0, 0, 0}, {0, 0, 0, 0, 0}};
+        initDimStrides(weights_dims, weights_strides, workspace);
+        return {filter, input, output, workspace};
+    }
+
+    return {filter, input, output, {}};
```

Comment:
And this is how construction problem can be resolved without `boost::optional`:
```suggestion
    StridedMemRef5D workspace{nullptr, nullptr, 0, {0, 0, 0, 0, 0}, {0, 0, 0, 0, 0}};
    {
        initDimStrides(weights_dims, weights_strides, workspace);
        return {filter, input, output, workspace};
    }

    return {filter, input, output, workspace};
```